swap_meet/vendor.py

```diff
@@ -1,2 +1,158 @@
+        for item in self.inventory:
+            if id == item.id:
+                return item
+    
+        return None
+    
+    # create instance method called swap_items- takes 3 arguments: other_vendor, my_item, their_item 
+    # return False if either inventory doesn't contain my_item or their_item
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+
+    # we can use the add and remove instance methods we created above for this function
+    # remove item from our inventory and add item to their inventory
+    # remove item from their inveory and add to our inventory.
+    # return True
+
+        self.remove(my_item)
+        other_vendor.add(my_item)
+        
+        other_vendor.remove(their_item)
+        self.add(their_item)
+
+        return True
+    
+    # create instance method called swap_first_item- takes one argument: other_vendor
+
+    def swap_first_item(self, other_vendor):
+
+        # if either inventory is empty, return False
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        
+        # remove first item in our inventory and add to their inventory
+        # remove first item in their inventory and add to our inventory
+        # return True
+
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+
+        return self.swap_items(other_vendor, my_first_item, their_first_item)
+    
+    # create method called get_by_category- takes in one argument: category (a string ex: 'Clothing')
+
+    def get_by_category(self, category):
+
+        # create empty list to place matching items
+
+        items_in_category = []
+
+        # iterate through each item in list
+
+        for item in self.inventory:
+            
+            # if category of that item matches category argument, add that item to the empty list
+
+            if category == item.get_category():
+                items_in_category.append(item)
+
+        # return the list 'items_in_category'
+        # if no items match the category, return empty list
+
+        return items_in_category
+    
+    # create instance method called get_best_by_category- takes in one argument: category (a string ex: 'Clothing')
+
+    def get_best_by_category(self, category):
+
+        # create variable to hold item with highest condition in matching category, assign as None
+        # initialize highest_condition variable at 0- will use to keep track of highest condition value
+        # create variable called list_items_in_category and use get_by_category method to create list of items that are of the same category
+
+        best_item = None
+        highest_condition = 0
+        list_items_in_cat = self.get_by_category(category)
+
+        # if no items of that category in list- return None
+
+        if list_items_in_cat == []:
+            return None
```

Comment:
This work, I’d like to offer a “pythonic” way to check for falsy values in Python" `if not list_items_in_cat`